lib/spack/spack/extensions.py

```diff
@@ -24,39 +24,47 @@ def extension_name(path):
         path (str): path where the extension resides
 
     Returns:
-        The extension name or None if path doesn't match the format
-        for Spack's extension.
+        The extension name. An exception is raised if path doesn't match
+        the expected format for a Spack command extension.
     """
-    regexp_match = re.search(extension_regexp, os.path.basename(path))
+    regexp_match = re.search(_extension_regexp, os.path.basename(path))
     if not regexp_match:
-        msg = "[FOLDER NAMING]"
-        msg += " {0} doesn't match the format for Spack's extensions"
-        tty.warn(msg.format(path))
-        return None
+        raise ExtensionNamingError(path)
     return regexp_match.group(1)
 
 
 def load_command_extension(command, path):
     """Loads a command extension from the path passed as argument.
 
     Args:
-        command (str): name of the command
+        command (str): name of the command (contains ``-``, not ``_``).
         path (str): base path of the command extension
 
     Returns:
-        A valid module object if the command is found or None
+        A valid module if found and loadable; None if not found. Module
+    loading exceptions are passed through.
     """
     extension = extension_name(path)
-    if not extension:
-        return None
 
     # Compute the name of the module we search, exit early if already imported
     cmd_package = '{0}.{1}.cmd'.format(__name__, extension)
+    # TODO: For consistency, the below conversion should utilize
+    #       spack.cmd.python_name(), but currently this would create a
+    #       circular relationship between spack.cmd and
+    #       spack.extensions.
```

Comment:
@alalazo @scheibelp Resolving this `TODO` would involve basically most of the file-based reorganization (without `find_commands()` and `get_command_module_from()`). I was planning to leave this to #11145 (or its successor), but I can resolve this here if you'd prefer.